test/library/draft/Optimizers/Optimizers.chpl

```diff
@@ -0,0 +1,303 @@
+    }
+
+    /* The current value the argument is using */
+    proc value {
+      // TODO: update when support multiple argument types
+      return intValue;
+    }
+
+    /* The lowest possible value the argument could be */
+    proc lowerBound {
+      // TODO: update when support multiple argument types
+      return intBounds(0);
+    }
+
+    /* The highest possible value the argument could be */
+    proc upperBound {
+      // TODO: update when support multiple argument types
+      return intBounds(1);
+    }
+
+    /* The number of possible values, when a discrete set of values was
+       specified */
+    proc setSize: int {
+      return this.setDom.size;
+    }
+
+    /* The domain of the set of possible values, when a discrete set of values
+       was specified */
+    proc setDom: domain(1) {
+      // TODO: update when support multiple argument types
+      return this.intDom;
+    }
+  }
+
+  /* Optimize using random sampling, returning the combination that leads to the
+     best result.
+
+     .. note::
+       Assumes the order of arguments in optimizableArgs is correct.  Cannot use
+       the name of the argument in the call (yet).
+
+       Assumes all `optimizableArgs` are used before all `args`.  Cannot
+       intermingle the two categories of arguments.
+
+     :arg func: The function to optimize.  This function is assumed to return a
+                result that is greater than or equal to zero.  Closer to zero
+                is assumed to be better.
+     :arg optimizableArgs: list of each optimizable aspect
+     :arg args: additional arguments for the function to use.  Expected to be
+                provided in a tuple, if provided
+
+     :returns: a list containing pairs of the name of each optimizable aspect
+               and the value found for it that leads to the best result for the
+               optimized function.
+   */
+  proc randomOptimize(func, optimizableArgs: list(optimizableArg),
+                      args:?t =none)
+    : list(optimizableArg) throws {
+
+    var basePoint = new Point(optimizableArgs);
+
+    var points: [0..<numIters] Point;
+    for i in points.indices {
+      points[i] = generateRandom(basePoint);
+    }
+
+    // Call the provided function with each possible set of values
+    if (args.type != nothing) {
+      evaluate(func, points, args);
+    } else {
+      evaluate(func, points);
+    }
+
+    var bestVal = max(real);
+    var bestIndex = -1;
+
+    for i in points.indices {
+      ref p = points[i];
+      if p.status == Status.completed && p.fom < bestVal {
+        bestVal = p.fom;
+        bestIndex = i;
+      }
+    }
+
+    var best: list(optimizableArg);
+
+    if bestIndex >= 0 {
+      for arg in points[bestIndex].parameters {
+        best.pushBack(arg);
+      }
+    } else {
+      // TODO: throw if nothing completed
+    }
+
+    return best;
+  }
+
+  config param randomSeed = 0;
+  private var rngInt = new randomStream(int, randomSeed);
```

Comment:
I had thought I remembered seeing that a seed of `0` meant "determine the seed yourself", but I must have gotten that from the HPO implementation rather than from the Random module, so I think you're right.  I'll look into making it actually random again, good catch!

Reply:
I made it so that a value of 0 would indicate not to use a specific seed.  But I could also have an additional param for determining whether to use the seed config at all, if you think that would be better?